src/Xamarin.Android.Build.Tasks/Utilities/MarshalMethodsAssemblyRewriter.cs

```diff
@@ -137,36 +138,23 @@ public void Rewrite (DirectoryAssemblyResolver resolver, List<string> targetAsse
 			// versions around.
 			foreach (string path in newAssemblyPaths) {
 				string? pdb = null;
-				string? mdb = null;
 
-				string source = Path.ChangeExtension (Path.Combine (Path.GetDirectoryName (path), Path.GetFileNameWithoutExtension (path)), ".pdb");
+				string source = Path.ChangeExtension (path, ".pdb");
 				if (File.Exists (source)) {
 					pdb = source;
 				}
 
-				source = $"{path}.mdb";
-				if (File.Exists (source)) {
-					mdb = source;
-				}
-
 				foreach (string targetPath in targetAssemblyPaths) {
-					string target = Path.Combine (targetPath, Path.GetFileNameWithoutExtension (path));
+					string target = Path.Combine (targetPath, Path.GetFileName (path));
 					CopyFile (path, target);
```

Comment:
I realize that the "original" version in 8bc7a3e84f didn't do this, but should we verify that *`target` exists* before this `CopyFile()`?

@grendello?

[`targetPaths` could contain *multiple* directories, such as `arm64-v8a/linked` and `x86_64/linked`](https://github.com/xamarin/xamarin-android/blob/dab495ff1e139e2761bd690cb001f3df78a8e0db/src/Xamarin.Android.Build.Tasks/Tasks/GenerateJavaStubs.cs#L240-L260), and it feels like this method could possibly copy/overwrite the "wrong" ABI?

Does `MarshalMethodAssemblyRewriter` properly deal with per-ABI assemblies?  (See also 929e7012410233e6814af369db582f238ba185ad?)

Reply:
@jonpryor all the known target-specific assemblies are ones which we won't rewrite, since they don't have any Java.Object-derived classes. But yes, in general, the code **would** fail if it encountered Java.Object-derived types in a target-specific assembly, since the task (and thus `MarshalMethodsAssemblyRewriter`) would be passed several copies of those assemblies.

Reply:
IOW, we're safe **now** with the code as-is, but it wouldn't hurt to amend it to make target-specific assemblies are handled properly. I guess the `DestinationSubPath` metadata item could be used to build the output path, by appending it to the `new/` path?

Reply:
@grendello: as per 929e7012410233e6814af369db582f238ba185ad, *any* assembly can now become a target-specific assembly if (when) it uses `IntPtr.Size`.  We likely shouldn't address this *here*, but we likely will need to address it.

Reply:
We should merge this PR, and followup later with a more complete fix for per-ABI assemblies.

Reply:
I agree, as it requires a review of all our tasks